### PR TITLE
Creating AddrIncoming from an existed TcpListener

### PR DESCRIPTION
**Pull request by zonyitoo**

### CODE REVIEW

File: src/server/tcp.rs [hunk 1]
```diff
@@ -51,6 +51,19 @@ impl AddrIncoming {
     pub fn bind(addr: &SocketAddr) -> crate::Result<Self> {
         AddrIncoming::new(addr)
     }
+    
+    /// Creates a new `AddrIncoming` from an existed `TcpListener`
+    pub fn from_listener(listener: TcpListener) -> crate::Result<Self> {
```
Comment (seanmonstar) [inline src/server/tcp.rs:56]: With this addition, `from_std` can be written to call it directly, ya?

Comment (zonyitoo) [reply src/server/tcp.rs:56]: Yes, of course.

File: src/server/tcp.rs [hunk 2]
```diff
@@ -36,6 +36,16 @@ impl AddrIncoming {
...
         let listener = TcpListener::from_std(std_listener).map_err(crate::Error::new_listen)?;
+        AddrIncoming::from_listener(listener)
+    }
+
+    /// Creates a new `AddrIncoming` binding to provided socket address.
+    pub fn bind(addr: &SocketAddr) -> crate::Result<Self> {
+        AddrIncoming::new(addr)
+    }
+
+    /// Creates a new `AddrIncoming` from an existed `TcpListener`
```
Comment (seanmonstar) [inline src/server/tcp.rs:47]:
```suggestion
    /// Creates a new `AddrIncoming` from an existing `tokio::net::TcpListener`.
```

**Comment by zonyitoo** [comment]
Great. When will be the next release?

**Pull request merged**